src/clj/puppetlabs/services/jruby_pool_manager/jruby_schemas.clj

```diff
@@ -127,8 +139,7 @@
                'JRubyInstanceState))
 
 (def JRubyPuppetInstanceInternal
-  {:flush-instance-fn IFn
-   :pool pool-queue-type
+  {:pool pool-queue-type
```

Comment:
Do we still need this? Can it instead be a reference to the "owning" pool-context? Would that make some of the signature changes unnecessary?

Reply:
I can look into how that is used. I know that we get a reference to the pool itself (the Java class) in a lot of places, but I think we pretty much always pull it out of `PoolState` above.

I think doing that would require more signature changes, while also still requiring the existing ones.. but maybe I'm missing which ones you're referring to?

Reply:
nvm, I was conflating a couple of things.